flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/sink/writer/AsyncSinkWriter.java

```diff
@@ -256,43 +242,54 @@ public AsyncSinkWriter(
             long maxTimeInBufferMS,
             long maxRecordSizeInBytes,
             Collection<BufferedRequestState<RequestEntryT>> states) {
+        this(
+                elementConverter,
+                context,
+                BasicAsyncSinkWriterConfiguration.builder()
+                        .setMaxBatchSize(maxBatchSize)
+                        .setMaxInFlightRequests(maxInFlightRequests)
+                        .setMaxBufferedRequests(maxBufferedRequests)
+                        .setMaxBatchSizeInBytes(maxBatchSizeInBytes)
+                        .setMaxTimeInBufferMS(maxTimeInBufferMS)
+                        .setMaxRecordSizeInBytes(maxRecordSizeInBytes)
+                        .build(),
+                states);
+    }
+
+    public AsyncSinkWriter(
+            ElementConverter<InputT, RequestEntryT> elementConverter,
+            Sink.InitContext context,
+            BasicAsyncSinkWriterConfiguration configuration,
```

Comment:
Adding to @dannycranmer comments. 

Why do we have this split between `BasicAsyncSinkWriterConfiguration` and `AsyncSinkWriterConfiguration`? If `BasicAsyncSinkWriterConfiguration` is supposed to be an internal class, then how users are supposed to construct `AsyncSinkWriter`?

If we indeed need/want `AsyncSinkWriterConfiguration`, then it seems to me like its builder and concrete implementation also should be publicly available class and we could basically squash them into one?

Reply:
Changed to AsyncSinkConfiguration implementation, and made both the builder and the configuration class public